chore: update changelog for version 2.13.1 (#10922)
DataDog/dd-trace-py: CHANGELOG.md

  - pymongo: Ensures instances of the `pymongo.MongoClient` can be patch after pymongo is imported

---

## 2.13.1

> ### New Features

Reviewer: nit: Could we alphabetize the sub-categories within the change types (e.g. within "Bug Fixes" change the ordering to Code Security > Profiling > Tracing)

> - Code Security (IAST): Always report a telemetry log error when an IAST propagation error raises, regardless of whether the \_DD_IAST_DEBUG environment variable is enabled or not.

Reviewer: ```suggestion
- Code Security: Always report a telemetry log error when an IAST propagation error raises, regardless of whether the `_DD_IAST_DEBUG` environment variable is enabled or not.
```

### Bug Fixes

- Tracing
  - celery: Fixes an issue where `celery.apply` spans didn't close if the `after_task_publish` or `task_postrun` signals didn't get sent when using `apply_async`, which can happen if there is an internal exception during the handling of the task. This update also marks the span as an error if an exception occurs.
  - celery: Fixes an issue where `celery.apply` spans using task_protocol 1 didn't close by improving the check for the task id in the body.
  - Removes a reference cycle that caused unnecessary garbage collection for top-level spans.

- Code Security

>   - fix potential memory leak on IAST exception handling.

Reviewer: ```suggestion
  - Fixes potential memory leak on IAST exception handling.
```

- Profiling

>   - all files with platform-dependent code have had their filenames updated to reflect the platform they are for. This fixes issues where the wrong file would be used on a given platform.

Reviewer: ```suggestion
  - Updates filenames for all files with platform-dependent code to reflect the platform they are for. This fixes issues where the wrong file would be used on a given platform.
```

  - Fixes endpoint profiling when using libdatadog exporter, either with `DD_PROFILING_EXPORT_LIBDD_ENABLED` or `DD_PROFILING_TIMELINE_ENABLED`.

>   - enables code provenance when using libdatadog exporter, `DD_PROFILING_EXPORT_LIBDD_ENABLED`, `DD_PROFILING_STACK_V2_ENABLED`, or `DD_PROFILING_TIMELINE_ENABLED`.

Reviewer: ```suggestion
  - Enables code provenance when using libdatadog exporter, `DD_PROFILING_EXPORT_LIBDD_ENABLED`, `DD_PROFILING_STACK_V2_ENABLED`, or `DD_PROFILING_TIMELINE_ENABLED`.
```

>   - fixes an issue where flamegraph was upside down for stack v2,  
>   `DD_PROFILING_STACK_V2_ENABLED`.

Reviewer: ```suggestion
  - Fixes an issue where the flame graph was upside down for stack v2 with `DD_PROFILING_STACK_V2_ENABLED`.
```
